solr/core/src/test/org/apache/solr/handler/RequestHandlerMetricsTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.handler;
+
+import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.handler.component.SearchHandler;
+import org.apache.solr.metrics.SolrDelegateRegistryMetricsContext;
+import org.apache.solr.request.SolrRequestHandler;
+import org.apache.solr.update.UpdateHandler;
+import org.junit.BeforeClass;
+
+/**
+ * Most of the tests for {@link org.apache.solr.handler.component.SearchHandler} are in {@link org.apache.solr.ConvertedLegacyTest}.
+ */
+public class RequestHandlerMetricsTest extends SolrTestCaseJ4 {
+  
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    initCore("solrconfig-aggreagatenodemetrics.xml", "schema.xml");
```

Comment:
The test in its current state tests the configuration of the metrics context instance variables.

If we really wanted to test that the functionality works as intended, we'd probably need a SolrCloud test (to get multiple cores over which to aggregate), testing that various metrics are aggregated as expected, and that non-aggregated metrics -- e.g. gauge -- are not reflected in aggregate metrics.

I think it'd be worth making the test(s) more robust, whether or not this is considered for potential upstream contribution, because these metrics are going to be used (of course!) for making decisions, and it might not be obvious in production use if aggregation's not actually working as intended. Worst-case scenario would be drawing conclusions/making decisions based on faulty metrics! (To be clear I think this looks good now, and agree that it should work. But it would be nice to be sure, particularly across future refactoring, forward-porting to 9.x/main, etc...)

Reply:
True. We should actually test the aggregate metrics are indeed returned correctly for a Solr instance with multiple cores

Reply:
I updated this to test actually hitting the metrics API. At this point I can only really test counts in the unit test since QTime can be off a bit from the times captured by these metrics making it hard to validated the aggregated request times in an automated way.